src/engine/renderer/tr_local.h

```diff
@@ -44,7 +44,10 @@ using i16vec4_t = int16_t[4];
 using u16vec4_t = uint16_t[4];
 using i16vec2_t = int16_t[2];
 using u16vec2_t = uint16_t[2];
-using f16vec4_t = int16_t[4]; // half float vector
+
+using f16_t = int16_t; // half float
+using f16vec2_t = i16vec2_t; // half float vector
+using f16vec4_t = i16vec4_t; // half float vector
```

Comment:
@slipher would you know another syntax that would error out if `i16vec4_t` is used instead of `f16vec4_t`?

Reply:
You could make f16_t a struct containing an int16_t. Maybe with a member function to convert to float

Reply:
There is no need for member function as we already have functions to convert back and forth to standard float.

I guess all we need is another type that has exact same size in memory than `int16_t` but that is not equal to `int16_t` when comparing types.

Can you write me that magical `f16_t` struct? 🙂️

Reply:
I pushed my attempt to the `slipher/f16` branch. I replicated most of this PR trying to get it to compile. There is one bit that does not compile at tr_vbo.cpp:402. This looks like a bug; I recall that @DolceTriade finding that the RSF_SPRITE flag makes images not show up so maybe that's why.

Reply:
So `vboData_t::spriteOrientation` is never set anywhere (meaning the faulty Vector4Copy is dead code), and it was that way since the field was introduced in b10e8c278c7817d52b6095f9f961598792a04dcb. I guess the "autosprite" thing (no idea what it is) has probably been broken at least since then. I guess we should delete the RSF_SPRITE stuff unless somebody has any idea what it is supposed to do.

Reply:
Thanks, I imported it over my changes, it builds and works. 👍️